include/triton/core/tritonbackend.h

```diff
@@ -1427,16 +1427,34 @@ TRITONSERVER_Error* TRITONBACKEND_ModelBatchIncludeRequest(
 /// \param model The backend model for which Triton is forming a batch.
 /// \param userp The placeholder for backend to store and retrieve information
 /// about this pending batch.
+/// \param cache_userp The read-only placeholder for backend to retrieve
+// information about the batching strategy for this model.
 /// \return a TRITONSERVER_Error indicating success or failure.
 TRITONSERVER_Error* TRITONBACKEND_ModelBatchInitialize(
-    TRITONBACKEND_Model* model, void** userp);
+    TRITONBACKEND_Model* model, void** userp, const void* cache_userp);
```

Comment:
Since you are adding the new APIs to store persistent batching info, you can wrap the "cache" in the way similar to [cache API](https://github.com/triton-inference-server/core/pull/142/files#diff-83cc1e14e9b115d1acaed238f93f014fc99fee507d0f467b0afd59669cb5bf4aR209-R238), I would call the implementation defined type to `TRITONBACKEND_Batcher` by the way. And because in the "new" function, you already pass `model` so that the implementation can retrieve information needed upfront. the signature for `TRITONBACKEND_ModelBatchInitialize` can be simplified to
```suggestion
    TRITONBACKEND_Batcher* batcher, void** userp);
```

Reply:
Thanks for this suggestion, Guan.

Are you suggesting replacing the (optional) InitializeCache functions with (mandatory) `TRITONBACKEND_BatcherNew`? Then, creating a separate `TRITONBACKEND_Batcher` struct that the user defines (but needs to store the model)?

Reply:
Thanks for chatting about this. Updated to use `TRITONBACKEND_Batcher` and make its initialization/finalization implementation required.